### PR TITLE
feat(WAF): add a new resource WAF rule anti crawler

### PR DESCRIPTION
**Pull request by deer-hang**
**What this PR does / why we need it**:
add a new resource WAF rule anti crawler
**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [X] Tests added/passed.
* [X] Documentation updated.
* [X] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/waf' TESTARGS='-run TestAccRuleAntiCrawler_'
...
==> Checking that code complies with gofmt requirements... 
TF_ACC=1 go test ./huaweicloud/services/acceptance/waf -v -run TestAccRuleAntiCrawler_ -timeout 360m -parallel 4 
=== RUN   TestAccRuleAntiCrawler_basic 
=== PAUSE TestAccRuleAntiCrawler_basic
=== RUN   TestAccRuleAntiCrawler_exceptProtectionMode
=== PAUSE TestAccRuleAntiCrawler_exceptProtectionMode
=== RUN   TestAccRuleAntiCrawler_withEpsID
=== PAUSE TestAccRuleAntiCrawler_withEpsID
=== CONT  TestAccRuleAntiCrawler_basic
=== CONT  TestAccRuleAntiCrawler_withEpsID
=== CONT  TestAccRuleAntiCrawler_exceptProtectionMode
--- PASS: TestAccRuleAntiCrawler_basic (348.15s) 
--- PASS: TestAccRuleAntiCrawler_withEpsID (350.22s) 
--- PASS: TestAccRuleAntiCrawler_exceptProtectionMode (352.06s) 
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/waf       352.137s
```

### CODE REVIEW

File: huaweicloud/services/waf/resource_huaweicloud_waf_rule_anti_crawler.go [hunk 1]
```diff
@@ -0,0 +1,349 @@
...
+				Optional:    true,
+				Description: `Specifies the reference table ID.`,
+			},
+		},
+	}
+}
+
+func updateAntiCrawlerProtectionMode(client *golangsdk.ServiceClient, d *schema.ResourceData,
+	cfg *config.Config) error {
+	updatePath := client.Endpoint + "v1/{project_id}/waf/policy/{policy_id}/anticrawler"
```
Comment (hmbig2) [inline huaweicloud/services/waf/resource_huaweicloud_waf_rule_anti_crawler.go]: those lines are the same as the lines in resourceRuleCreate

File: docs/resources/waf_rule_anti_crawler.md [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+* `id` - The resource ID.
+
+* `status` - The rule status.
+
+## Import
+
+There are two ways to import WAF rule anti crawler state.
+
+* Using `policy_id` and `rule_id`, separated by a slash, e.g.
```
Comment (hmbig2) [inline docs/resources/waf_rule_anti_crawler.md:101]:
```suggestion
* Using `policy_id` and `id`, separated by a slash, e.g.
```

File: huaweicloud/services/waf/resource_huaweicloud_waf_rule_anti_crawler.go (hunk 1 above)
Comment (deer-hang) [reply huaweicloud/services/waf/resource_huaweicloud_waf_rule_anti_crawler.go]: OK, I have removed the duplicate codes.

File: docs/resources/waf_rule_anti_crawler.md (hunk 2 above)
Comment (deer-hang) [reply docs/resources/waf_rule_anti_crawler.md:101]: Consistent with other rule resources.

**Pull request merged**